002-add_two_numbers/main_3.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Definition for singly-linked list.
+ * struct ListNode {
+ *     int val;
+ *     ListNode *next;
+ *     ListNode() : val(0), next(nullptr) {}
+ *     ListNode(int x) : val(x), next(nullptr) {}
+ *     ListNode(int x, ListNode *next) : val(x), next(next) {}
+ * };
+ */
+class Solution {
+public:
+    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
+        ListNode dummy = ListNode(0);
+        ListNode* result = &dummy;
+        bool carry = false;
+        
+        while (l1 || l2) {
+            int val = 0;
+            if (l1) {
+                val += l1->val;
+            }
+            if (l2) {
+                val += l2->val;
+            }
+            if (carry) {
+                val++;
+            }
+            carry = val >= 10;
+            if (carry) {
+                val -= 10;
+            }
+            result->next = new ListNode(val);
```

Comment:
```suggestion
            result->next = new ListNode(val % 10);
```

とすると、

```
            if (carry) {
                val -= 10;
            }
```
を丸ごと消せそうですね